20w14infinite+1.2.1 - attempt to fix stopWorkers deadlock

tildejustin opened the pull request.

jellysquid3 committed:
fix: Send chunk unload events when load distance changes

Closes #673

jellysquid3 committed:
fix: Track chunk map re-centering more carefully

tildejustin committed:
attempt to fix deadlock caused by rare wait - notifyAll race condition when worker threads are stopped

tildejustin committed:
bump version to 1.2.1

DuncanRuns reviewed (approved):
(See https://github.com/Minecraft-Java-Edition-Speedrunning/sodium/pull/12#pullrequestreview-2069665246 for comment about deadlock fix)

tildejustin committed:
change: Patch vanilla chunk manager instead of rolling our own

The custom hash table based implementation was not sound. This might also
fix a few issues with mod compatibility. Should still be faster than the
vanilla implementation, and it will significantly reduce the amount of code
we're maintaining.

DuncanRuns reviewed (approved):
(Same stuff as https://github.com/Minecraft-Java-Edition-Speedrunning/sodium/pull/12#pullrequestreview-2069665246)
